### DISCUSSION TITLE
Renaming Document Titles from Filename?

### DISCUSSION THREAD

**Discussion by ryanwwest** [koreader/koreader#11074] [Q&A] [answered]
I often have a bunch of scanned or otherwise unlabeled PDFs that have a descriptive filename but no metadata, such as a title. But I also have tons of PDFs where the title is more descriptive than the filename. So neither File Browser option works for all of my library. I haven't seen a way to display both the title and filename together, which would be a good workaround. But I know @hius07's custom metadata is a way to overwrite the title and display something new, so my best option is individually adding a custom title to each document that matches the filename (without extension). I was wondering if a faster way exists to do this for quickly renaming a title to be the filename (no copy-paste necessary) or some sort of way to do this in batch?

**Comment by Frenzie** [comment, marked as answer]
> I often have a bunch of scanned or otherwise unlabeled PDFs that have a descriptive filename but no metadata, such as a title. But I also have tons of PDFs where the title is more descriptive than the filename. So neither File Browser option works for all of my library.

As you wrote it I don't see any problem. ;-)

Perhaps you meant to say they have junk metadata rather than no metadata?

Edit: in any case, you can ignore metadata on a per-file basis.

**Reply by ryanwwest** [reply]
I guess their title metadata could be considered junk metadata, the title was 'downloadFile'. Ignoring metadata is just what I needed, as it then displays the filename as title with just two taps per file. Does this also ignore TOC or is that not metadata? Thanks!

This is a good enough solution for me. It would be nicer to select a range of files to 'ignore metadata' but I'm not aware of range/batch selections in the File Browser anyway so I don't think we need to worry about it. If that ever gets implemented, this is one use case.

**Reply by Frenzie** [reply]
> the title was 'downloadFile'

The very definition. ;-)

> Does this also ignore TOC or is that not metadata?

That's metadata but that's irrelevant to browsing files.

**Comment by Frenzie** [comment]
Use the plus in the top right.

**Reply by ryanwwest** [reply]
Got it. I wonder if it would be easy or useful to others to allow the ignore Metadata action to work for selections, not just individual files.

**Reply by Frenzie** [reply]
That's probably a question for @hius07 :-P
